### PR TITLE
create PDP fragment

### PR DESCRIPTION
**Pull request by tolia-ogurt**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,89 +1,10 @@
-# Built application files
-*.apk
```
Comment (DaryaErofeeva) [inline .gitignore]: Move these changes to Initial commit on master

File: app/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+/build
```
Comment (DaryaErofeeva) [inline app/.gitignore]: We should have only one `.gitignore` for the whole project

File: app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    compileSdk 31
+
+    defaultConfig {
+        applicationId "com.example.trempel"
+        minSdk 21
+        targetSdk 31
+        versionCode 1
+        versionName "1.0"
+
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
```
Comment (DaryaErofeeva) [inline app/build.gradle]: Have you already added any tests?

File: app/build.gradle [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+dependencies {
+
+    implementation 'androidx.core:core-ktx:1.7.0'
+    implementation 'androidx.appcompat:appcompat:1.3.1'
+    implementation 'com.google.android.material:material:1.4.0'
+    implementation 'androidx.constraintlayout:constraintlayout:2.1.1'
+    testImplementation 'junit:junit:4.+'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.3'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
```
Comment (DaryaErofeeva) [inline app/build.gradle]: I don't think we need these test dependencies for now, please remove them

File: app/src/androidTest/java/com/example/trempel/ExampleInstrumentedTest.kt [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ */
+@RunWith(AndroidJUnit4::class)
+class ExampleInstrumentedTest {
+    @Test
+    fun useAppContext() {
+        // Context of the app under test.
+        val appContext = InstrumentationRegistry.getInstrumentation().targetContext
+        assertEquals("com.example.trempel", appContext.packageName)
+    }
+}
```
Comment (DaryaErofeeva) [inline app/src/androidTest/java/com/example/trempel/ExampleInstrumentedTest.kt]: Let's remove this automatically generated test class

File: app/build.gradle [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        versionCode 1
+        versionName "1.0"
+
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
```
Comment (DaryaErofeeva) [inline app/build.gradle]: I think we can get rid of proguard now

File: app/proguard-rules.pro [hunk 7]
```diff
@@ -0,0 +1,21 @@
+# Add project specific ProGuard rules here.
```
Comment (DaryaErofeeva) [inline app/proguard-rules.pro]: I think we can get rid of proguard now, for now this's just the file bunch of comments

File: app/src/main/java/com/example/trempel/PdpFragment.kt [hunk 8]
```diff
@@ -0,0 +1,7 @@
+package com.example.trempel
+
+import androidx.fragment.app.Fragment
+
+class PdpFragment:Fragment(R.layout.pdp_fragment) {
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/PdpFragment.kt]: Please, follow the formatting convention suggested by JetBrains https://kotlinlang.org/docs/coding-conventions.html#colon
`Always put a space after :.`

_You can use `Command + Alt + L` and it'll do some formatting magic for you_

File: app/src/test/java/com/example/trempel/ExampleUnitTest.kt [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+class ExampleUnitTest {
+    @Test
+    fun addition_isCorrect() {
+        assertEquals(4, 2 + 2)
+    }
+}
```
Comment (DaryaErofeeva) [inline app/src/test/java/com/example/trempel/ExampleUnitTest.kt]: Let's remove this automatically generated test class

File: app/src/main/res/layout/activity_main.xml [hunk 10]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (DaryaErofeeva) [inline app/src/main/res/layout/activity_main.xml]: Do we actually need ConstraintLayout as a wrapper in this case?

File: app/src/main/res/layout/pdp_fragment.xml [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+        android:layout_marginTop="@dimen/normal_150"
+        android:layout_marginEnd="@dimen/large_250"
+        android:scaleType="centerInside"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        android:src="@drawable/image_product" />
+
+    <View
+        android:id="@+id/cardView"
```
Comment (DaryaErofeeva) [inline app/src/main/res/layout/pdp_fragment.xml]: why do we have `@+id/cardView` value for id?

File: app/src/main/res/layout/pdp_fragment.xml [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+        android:text="@string/product_description"
+        android:textColor="@color/white"
+        app:layout_constraintBottom_toTopOf="@id/btnAddToCart"
+        app:layout_constraintTop_toBottomOf="@id/productScore"
+        app:layout_constraintStart_toStartOf="@id/cardView"
+        app:layout_constraintEnd_toEndOf="@id/cardView"
+        />
+
+    <Button
+        android:id="@+id/btnAddToCart"
```
Comment (DaryaErofeeva) [inline app/src/main/res/layout/pdp_fragment.xml]: You have 3 naming approaches in this file:
* `iV_product` - abbreviation of view type  + `_` + `product`
* `productTitle `, `productPrice`, `productDescription` etc. - `product` + content type
* `btnAddToCart` - abbreviation of view type + view text

Let's stick to one of them, I'd suggest `abbreviation + content type` (e.g. `iv_product_image`, `tv_product_title` etc.) but feel free to apply other option as the main goal is naming consistency

File: app/src/main/res/layout/pdp_fragment.xml [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+        android:layout_width="0dp"
+        android:layout_height="56dp"
```
Comment (DaryaErofeeva) [inline app/src/main/res/layout/pdp_fragment.xml]: Should be extracted as dimention

File: app/src/main/res/values-night/themes.xml [hunk 14]
```diff
@@ -0,0 +1,16 @@
+<resources xmlns:tools="http://schemas.android.com/tools">
+    <!-- Base application theme. -->
+    <style name="Theme.Trempel" parent="Theme.MaterialComponents.DayNight.DarkActionBar">
```
Comment (DaryaErofeeva) [inline app/src/main/res/values-night/themes.xml]: The additional task will be created for changing the theme colors

File: app/src/main/res/values/strings.xml [hunk 15]
```diff
@@ -0,0 +1,11 @@
+<resources>
+    <string name="app_name">Trempel</string>
+    <string name="product_title">Mens Casual Premium Slim Fit T-Shirts</string>
+    <string name="product_price">$22.3</string>
+    <string name="product_score">5</string>
+    <string name="product_reviews">(228 reviews)</string>
+    <string name="product_description"
+        >"Slim-fitting style, contrast raglan long sleeve, three-button henley placket, light weight soft fabric for breathable and comfortable wearing. \nAnd Solid stitched shirts with round neck made for durability and a great fit for casual fashion wear and…"
```
Comment (DaryaErofeeva) [inline app/src/main/res/values/strings.xml]: Something happened to the formatting

File: app/src/main/res/values/themes.xml [hunk 16]
```diff
@@ -0,0 +1,16 @@
+<resources xmlns:tools="http://schemas.android.com/tools">
```
Comment (DaryaErofeeva) [inline app/src/main/res/values/themes.xml:1]: I don't think we should have light and dark theme

File: app/src/main/java/com/example/trempel/PdpFragment.kt [hunk 17]
```diff
@@ -0,0 +1,7 @@
...
+
+}
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/PdpFragment.kt:19]: I don't think we need class body here

File: app/src/main/res/layout/pdp_fragment.xml [hunk 18]
```diff
@@ -0,0 +1,144 @@
...
+    <TextView
+        android:id="@+id/productTitle"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/normal_150"
+        android:layout_marginTop="@dimen/normal_150"
+        android:layout_marginEnd="@dimen/normal_150"
+        android:fontFamily="@font/montserrat_black"
+        android:textColor="@color/white"
+        android:textSize="18sp"
```
Comment (DaryaErofeeva) [inline app/src/main/res/layout/pdp_fragment.xml]: textSizes should also be extracted from the layout file

File: app/src/main/res/layout/pdp_fragment.xml [hunk 19]
```diff
@@ -0,0 +1,144 @@
...
+
+    <TextView
+        android:id="@+id/productPrice"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/normal_150"
+        android:layout_marginEnd="@dimen/normal_150"
+        android:font="@font/montserrat_light"
+        android:textColor="@color/white"
+        android:textSize="24sp"
```
Comment (DaryaErofeeva) [inline app/src/main/res/layout/pdp_fragment.xml]: textSizes should also be extracted from the layout file

File: app/src/main/res/layout/pdp_fragment.xml [hunk 20]
```diff
@@ -0,0 +1,144 @@
...
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        app:layout_constraintTop_toBottomOf="@id/iV_product"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        android:background="@drawable/ic_rectangle_1"/>
+
+    <TextView
+        android:id="@+id/productTitle"
```
Comment (DaryaErofeeva) [inline app/src/main/res/layout/pdp_fragment.xml]: Looks like you haven't defined the lineHeight

File: app/src/main/res/layout/pdp_fragment.xml [hunk 21]
```diff
@@ -0,0 +1,144 @@
...
+        android:textColor="@color/white"
+        android:textSize="18sp"
+        app:layout_constraintEnd_toStartOf="@id/productPrice"
+        app:layout_constraintStart_toStartOf="@id/cardView"
+        app:layout_constraintTop_toTopOf="@id/cardView"
+        android:text="@string/product_title"
+        />
+
+    <TextView
+        android:id="@+id/productPrice"
```
Comment (DaryaErofeeva) [inline app/src/main/res/layout/pdp_fragment.xml]: Looks like you haven't defined the lineHeight

File: app/src/main/res/layout/pdp_fragment.xml [hunk 22]
```diff
@@ -0,0 +1,144 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@color/white">
+
+    <ImageView
+        android:id="@+id/iV_product"
+        android:layout_width="0dp"
+        android:layout_height="272dp"
```
Comment (DaryaErofeeva) [inline app/src/main/res/layout/pdp_fragment.xml]: This value should also be extracted to dimentions

**Review by DaryaErofeeva — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (tolia-ogurt) [reply .gitignore]: ok

**Review by DaryaErofeeva — APPROVED** [review]

**Pull request merged**